aiven/resource_flink_test.go

```diff
@@ -0,0 +1,240 @@
+			{
+				Config: fmt.Sprintf(`
+variable "project_name" {
+  type = string
+  default = "%s"
+}
+variable "service_name" {
+  type = string
+  default = "%s"
+}
+
+resource "aiven_flink" "bar" {
+  project = var.project_name
+  cloud_name = "google-europe-west1"
+  plan = "business-4"
+  service_name = var.service_name
+  maintenance_window_dow = "monday"
+  maintenance_window_time = "10:00:00"
+
+  flink_user_config {
+    number_of_task_slots = 10
+    parallelism_default = 2
+    restart_strategy = "failure-rate"
+  }
+}
+
+data "aiven_flink" "service" {                                                                  
+  service_name = aiven_flink.bar.service_name                                                   
+  project = aiven_flink.bar.project                                                             
+}                                                                      
+`,
+					serviceName,
+					projectName,
+				),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAivenServiceCommonAttributes("data.aiven_flink.service"),
+					resource.TestCheckResourceAttr(resourceName, "project", projectName),
+					resource.TestCheckResourceAttr(resourceName, "service_name", serviceName),
+					resource.TestCheckResourceAttr(resourceName, "state", "RUNNING"),
+					resource.TestCheckResourceAttr(resourceName, "service_type", "flink"),
+					resource.TestCheckResourceAttr(resourceName, "cloud_name", "google-europe-west1"),
+					resource.TestCheckResourceAttr(resourceName, "maintenance_window_dow", "monday"),
+					resource.TestCheckResourceAttr(resourceName, "maintenance_window_time", "10:00:00"),
+					resource.TestCheckResourceAttr(resourceName, "termination_protection", "false"),
+					resource.TestCheckResourceAttr(resourceName, "flink_user_config.0.number_of_task_slots", "10"),
+					resource.TestCheckResourceAttr(resourceName, "flink_user_config.0.parallelism_default", "2"),
+					resource.TestCheckResourceAttr(resourceName, "flink_user_config.0.restart_strategy", "failure-rate"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAiven_flinkKafkaToKafka(t *testing.T) {
+
+	randString := func() string { return acctest.RandStringFromCharSet(10, acctest.CharSetAlpha) }
+
+	projectName := os.Getenv("AIVEN_PROJECT_NAME")
+	flinkServiceName := fmt.Sprintf("test-acc-flink-%s", randString())
+	kafkaServiceName := fmt.Sprintf("test-acc-flink-kafka-%s", randString())
+	sourceTopicName := fmt.Sprintf("test-acc-flink-kafka-source-topic-%s", randString())
+	sinkTopicName := fmt.Sprintf("test-acc-flink-kafka-sink-topic-%s", randString())
+	sourceTableName := fmt.Sprintf("test_acc_flink_kafka_source_table_%s", randString())
+	sinkTableName := fmt.Sprintf("test_acc_flink_kafka_sink_table_%s", randString())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckAivenServiceResourceDestroy,
```

Comment:
Here we need to check if a Job and Table are destroyed.

Reply:
done